frontend/server/src/Controllers/Course.php

```diff
@@ -4242,26 +4257,48 @@ public static function getArenaCourseDetailsForTypeScript(
             /*required=*/false,
             fn (string $alias) => \OmegaUp\Validators::alias($alias),
         );
-
         if (is_null($problemAlias)) {
             return $response;
         }
 
+        $problemset = \OmegaUp\DAO\Problemsets::getByPK(
+            intval($assignment->problemset_id)
+        );
+        if (is_null($problemset)) {
+            throw new \OmegaUp\Exceptions\NotFoundException(
+                'problemsetNotFound'
+            );
+        }
+
         $problem = \OmegaUp\DAO\Problems::getByAliasAndProblemset(
             $problemAlias,
-            intval($assignment->problemset_id)
+            intval($problemset->problemset_id)
         );
         if (is_null($problem)) {
             throw new \OmegaUp\Exceptions\NotFoundException(
                 'problemNotFound'
             );
         }
 
-        $response['smartyProperties']['payload']['currentProblem'] = [
-            'alias' => strval($problem->alias),
-            'title' => strval($problem->title),
-            // TODO: Add more information about the currentProblem
-        ];
+        $problemDetails = \OmegaUp\Controllers\Problem::getProblemDetails(
+            $r->identity,
+            $problem,
+            $problemset,
+            \OmegaUp\Controllers\Identity::getPreferredLanguage(
+                $r->identity,
+            ),
+            /*showSolvers=*/false,
+            /*preventProblemsetOpen=*/false,
+            /*contestAlias=*/null,
```

Comment:
ahora podemos usar los nombres de los campos!

```suggestion
            showSolvers: false,
            preventProblemsetOpen: false,
            contestAlias: null,
```